File: src/main/java/fr/vlegall/sochief/model/common/DurationMillisConverter.java
package fr.vlegall.sochief.model.common;

import jakarta.persistence.AttributeConverter;
import jakarta.persistence.Converter;

import java.time.Duration;

@Converter(autoApply = true)
public class DurationMillisConverter implements AttributeConverter<Duration, Long> {
    @Override
    public Long convertToDatabaseColumn(Duration attribute) {
        return attribute != null ? attribute.toMillis() : null;
    }

    @Override
    public Duration convertToEntityAttribute(Long dbData) {
        return dbData != null ? Duration.ofMillis(dbData) : null;
    }
}

File: src/main/kotlin/com/vlegall/sochief/model/common/DurationMillisConverter.kt
package com.vlegall.sochief.model.common

import jakarta.persistence.AttributeConverter
import jakarta.persistence.Converter
import java.time.Duration


@Converter(autoApply = true)
class DurationMillisConverter : AttributeConverter<Duration, Long> {
    override fun convertToDatabaseColumn(attribute: Duration?): Long? =
        attribute?.toMillis()

    override fun convertToEntityAttribute(dbData: Long?): Duration? =
        dbData?.let { Duration.ofMillis(it) }
}